src/apiPort.js

```diff
@@ -235,10 +235,15 @@ class ApiPort {
       for (const testDataPath of lookIn) {
         if (testDataPath) {
           const filePath = `${testDataPath}/${fileName}.data`
-          const fileData = loadFile(filePath)
-          if (fileData) {
-            this.apiPort.$file[fileName] = fileData
-            break
+          try {
+            const fileData = loadFile(filePath)
+            if (fileData) {
+              this.apiPort.$file[fileName] = fileData
+              break
+            }
+          } catch (e) {
+            delete this.apiPort.$file[fileName]
+            // Ignored here if not found will re-throw the error in the end of this loop
```

Comment:
This is not quite a re-throw. What if the exception is bad yaml?

Probably should add some unit tests to the ApiPort to test this. Plus, this code is using exceptions to determine the status of file existence - which is why it might mask the real error. It think it would be better to do an exist check before trying to load.

Reply:
The re-trow is in the line https://github.com/testbetter/openapitest/blob/51559ac4d4bc74c7c729f0777327993064a47fda/src/apiPort.js#L252 after the loop. I agree we should check for the file existence here. But for this small bug fix, I would not like to change this logic too much.  If a corrupt `YAML` file is passed it will result in an error like `Cannot find test data file: ${fileName}.data.(js|yaml) in any of...`.  Which I agree is not the best way to express it, I can fix it here or in a future PR where we include more unit tests for this class.  If you think I should fix it in this pull request, I can push the code change by tomorrow.

Reply:
Fixing it later is OK. I can create an issue here to track it. For anyone who runs across this error, it is very misleading since they will believe the error first (since who ever writes buggy YAMLs right?) and spend some time trying to figure out why the "perfectly fine" file they wrote is not being found by the framework.